k8s/k8s.go

```diff
@@ -316,6 +445,29 @@ func overrideIPAMResult(ipAddrsNoIpam string, logger *log.Entry) (*types.Result,
 	return &result, nil
 }
 
+// parseIPAddrs is a utility function that parses string of IPs in json format that are
+// passed in as a string and returns a slice of string with IPs.
+// It also makes sure the slice isn't empty.
+func parseIPAddrs(annotation, ipAddrsStr string, logger *log.Entry) ([]string, error) {
```

Comment:
If we log the annotation before calling this then it's moot, and makes this code much better encapsulated than it is now.

The logs would look something like this:

```
INFO Parsing annotation 'cni.projectcalico.org/ipAddrs': [..., ...]
ERROR Failed to parse [..., ...] as json: <error msg here>
```

We don't need to log out the annotation in every log - the context comes from the logs surrounding it.